components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests

```diff
@@ -65,12 +68,25 @@ if [ "$S3_DATA_BUCKET" == "" ]; then
   exit 1
 fi
 
+if [[ "$SKIP_FSX_TESTS" == "false" && "$EKS_PRIVATE_SUBNETS" == "" ]]; then
+  echo "Missing EKS private subnets"
+  usage
+  exit 1
+fi
+
 function cleanup() {
   set +e
 
   cleanup_kfp
   delete_generated_role
 
+  if [[ "${SKIP_FSX_TESTS}" == "false" ]]; then
+    delete_fsx_instance
+    # Sleep in order for the security group to detach before attempting to delete it
+    sleep 15s
```

Comment:
is there a better condition based solution for this?

Reply:
I have not had much of a chance to play around with this. I think maybe once the FSx file system changes status to "Deleting" it would have detached from the VPC and security groups. Not entirely sure, this was a transient error.